scheduling.hpp

```diff
@@ -59,13 +62,39 @@ enum class StepUnit {
     Day, Week, Month
 };
 
+/**
+ * @brief Get step enum from string
+
+ * Throws an std::invalid_argument exception if the value
+ * is not supported.
+ */
+inline StepUnit step_unit_enum_from_string(const std::string& text)
+{
+    std::map<std::string, StepUnit> mapping{
+        {"day", StepUnit::Day},
+        {"week", StepUnit::Week},
+        {"month", StepUnit::Month}
+    };
+    try {
+        return mapping.at(text);
+    }
+    catch (const std::out_of_range&) {
+        throw std::invalid_argument("step_unit_enum_from_string:"
+                                    " Invalid value '" + text +"' provided");
+    }
+}
+
+
+
 class Scheduler
 {
 public:
     /**
      * Scheduler creates a vector of simulation steps
      * based on start, end date, unit and number of units.
      * The steps can be e.g. 1 day, 3 months, 2 weeks.
+     * Step is scheduled if start date of an interval is
+     * within [start, end] even if end date of the step is outside.
```

Comment:
This describes behavior, but please add why it is useful, i.e., "...so that the whole requested interval is contained in the resulting steps..." or sth like that.